syntax = "proto3";

import public "common.proto";

package img_common;

message CameraInfo { string cameraId = 1; }

enum ImgType {
  Raw = 0;
  DebugLandmarks = 1;
}

message Image {
  int32 width = 1;
  int32 height = 2;
  int32 channels = 3;
  // RGB (mediapipe supported)
  bytes data = 4;
  ImgType type = 5;
}

message CameraFrame {
  .common.MsgHeader header = 1;
  CameraInfo cameraInfo = 2;
  Image image = 3;
}
